src/ripple/app/misc/impl/Manifest.cpp

```diff
@@ -309,6 +309,30 @@ ManifestCache::getMasterKey (PublicKey const& pk) const
     return pk;
 }
 
+std::uint32_t
+ManifestCache::getSequence (PublicKey const& pk) const
+{
+    std::lock_guard lock{read_mutex_};
+    auto const iter = map_.find (pk);
+
+    if (iter != map_.end () && !iter->second.revoked ())
+        return iter->second.sequence;
+
+    return 0;
+}
+
+std::string
```

Comment:
This and `ManifestCache::getSequence` should probably wrap their return in a `boost::optional`, making it possible to distinguish a "I can't find the key you asked for" from "the key you asked for doesn't have the attribute you request".

Reply:
Done